src/vs/workbench/browser/parts/quickopen/quickOpenController.ts

```diff
@@ -329,7 +329,7 @@ export class QuickOpenController extends Component implements IQuickOpenService
 		}
 
 		// Respect input value
-		if (options.value) {
+		if (options.value && !options.valueSelection) {
 			this.pickOpenWidget.setValue(options.value, options.valueSelection);
```

Comment:
Aren't we loosing the value here as soon as a selection is provided? Isn't the better change to reset `valueSelection` after an validation pass?

Reply:
Maybe here: https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/browser/parts/quickopen/quickOpenController.ts#L209

Reply:
The issue I found (just to clarify my thinking, not implying that it's correct) is that we set the value if it's present in options.value, which causes the cursor to move to the end of the string. 

By checking that we haven't already started the validation process (init), we can set the value / placeholder / template value, and then when the user begins to enter their own (and thus we start validating against the rule set), and ignore setting the value itself which would cause the cursor jump. I'm unsure how the selection works in all it's entirety, so if you have a better solution / idea I'm open to fixing this PR and learning from the updated fix.

Reply:
> which causes the cursor to move to the end of the string.

I believe that happens because we explicitly set the *value selection*